tests/baselines/reference/arrowFunctionReturnTypeErrorSpan.errors.txt

```diff
@@ -0,0 +1,28 @@
+arrowFunctionReturnTypeErrorSpan.ts(3,3): error TS2322: Type 'string' is not assignable to type 'number'.
+arrowFunctionReturnTypeErrorSpan.ts(12,25): error TS2304: Cannot find name 'missing'.
+arrowFunctionReturnTypeErrorSpan.ts(7,15): error TS2322: Type 'string' is not assignable to type 'number'.
+arrowFunctionReturnTypeErrorSpan.ts(10,15): error TS2322: Type 'string' is not assignable to type 'number'.
+
+
+==== arrowFunctionReturnTypeErrorSpan.ts (4 errors) ====
+    // block body
+    const a = (): number => {
+      return "foo";
+      ~~~~~~
+!!! error TS2322: Type 'string' is not assignable to type 'number'.
+    };
+    
+    // expression body
+    const b = (): number => "foo";
+                  ~~~~~~
+!!! error TS2322: Type 'string' is not assignable to type 'number'.
+    
+    type F<T> = T;
+    const c = (): F<number> => "foo";
+                  ~
+!!! error TS2322: Type 'string' is not assignable to type 'number'.

```

Comment:
It would probably be better to highlight `F<number>` rather than just `F`, but I'm not sure how to do that?

Reply:
I believe this patch solves the issue 🤔 
```diff
diff --git a/src/compiler/utilities.ts b/src/compiler/utilities.ts
index 2ea2c5ee56..a2ed642e5f 100644
--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -2531,7 +2531,7 @@ export function getErrorSpanForNode(sourceFile: SourceFile, node: Node): TextSpa
         const arrowFn = node.parent as ArrowFunction;
         if (arrowFn.body === node && arrowFn.type !== undefined) {
             const pos = skipTrivia(sourceFile.text, arrowFn.type.pos);
-            return getSpanOfTokenAtPosition(sourceFile, pos);
+            return createTextSpanFromBounds(pos, arrowFn.type.end);
         }
     }
 
```

after running tests I get these diffs:

```diff
--- tests/baselines/reference/conditionalTypes1.errors.txt	2024-12-17 17:30:42.931108593 -0500
+++ tests/baselines/local/conditionalTypes1.errors.txt	2024-12-17 17:31:14.443317916 -0500
@@ -466,5 +466,5 @@
     const f44 = <U>(value: T94<U>): T95<U> => value;
     const f45 = <U>(value: T95<U>): T94<U> => value;  // Error
-                                    ~~~
+                                    ~~~~~~
 !!! error TS2322: Type 'T95<U>' is not assignable to type 'T94<U>'.
 !!! error TS2322:   Type 'number | boolean' is not assignable to type 'T94<U>'.

```
and
```diff
--- tests/baselines/reference/arrowFunctionReturnTypeErrorSpan.errors.txt	2024-12-17 17:21:00.321730450 -0500
+++ tests/baselines/local/arrowFunctionReturnTypeErrorSpan.errors.txt	2024-12-17 17:34:54.015871276 -0500
@@ -1,7 +1,7 @@
 arrowFunctionReturnTypeErrorSpan.ts(3,3): error TS2322: Type 'string' is not assignable to type 'number'.
-arrowFunctionReturnTypeErrorSpan.ts(12,25): error TS2304: Cannot find name 'missing'.
 arrowFunctionReturnTypeErrorSpan.ts(7,15): error TS2322: Type 'string' is not assignable to type 'number'.
 arrowFunctionReturnTypeErrorSpan.ts(10,15): error TS2322: Type 'string' is not assignable to type 'number'.
+arrowFunctionReturnTypeErrorSpan.ts(12,15): error TS2304: Cannot find name 'missing'.
 
 
 ==== arrowFunctionReturnTypeErrorSpan.ts (4 errors) ====
@@ -19,10 +19,10 @@
     
     type F<T> = T;
     const c = (): F<number> => "foo";
-                  ~
+                  ~~~~~~~~~
 !!! error TS2322: Type 'string' is not assignable to type 'number'.
     
     const d = (): number => missing;
-                            ~~~~~~~
+                  ~~~~~~
 !!! error TS2304: Cannot find name 'missing'.
     
\ No newline at end of file
```